plotly/tools.py

```diff
@@ -1451,6 +1455,823 @@ class FigureFactory(object):
+
+        # plot
+        py.iplot(fig, filename='Violin Plot')
+        ```
+
+        Example 2: Multiple Violin Plots with Qualitative Coloring
+        ```
+        import plotly.plotly as py
+        from plotly.tools import FigureFactory as FF
+        from plotly.graph_objs import graph_objs
+
+        import numpy as np
+        import pandas as pd
+        from scipy import stats
+
+        # create dataframe
+        np.random.seed(619517)
+        Nr=250
+        y = np.random.randn(Nr)
+        gr = np.random.choice(list("ABCDE"), Nr)
+        norm_params=[(0, 1.2), (0.7, 1), (-0.5, 1.4), (0.3, 1), (0.8, 0.9)]
+
+        for i, letter in enumerate("ABCDE"):
+            y[gr == letter] *=norm_params[i][1]+ norm_params[i][0]
+        df = pd.DataFrame(dict(Score=y, Group=gr))
+
+        # create violin fig
+        fig = FF.create_violin(df, data_header='Score', group_header='Group',
+                               height=600, width=1000)
+
+        # plot
+        py.iplot(fig, filename='Violin Plot with Coloring')
+        ```
+
+        Example 3: Violin Plots with Colorscale
+        ```
+        import plotly.plotly as py
+        from plotly.tools import FigureFactory as FF
+        from plotly.graph_objs import graph_objs
+
+        import numpy as np
+        import pandas as pd
+        from scipy import stats
+
+        # create dataframe
+        np.random.seed(619517)
+        Nr=250
+        y = np.random.randn(Nr)
+        gr = np.random.choice(list("ABCDE"), Nr)
+        norm_params=[(0, 1.2), (0.7, 1), (-0.5, 1.4), (0.3, 1), (0.8, 0.9)]
+
+        for i, letter in enumerate("ABCDE"):
+            y[gr == letter] *=norm_params[i][1]+ norm_params[i][0]
+        df = pd.DataFrame(dict(Score=y, Group=gr))
+
+        # define header params
+        data_header = 'Score'
+        group_header = 'Group'
+
+        # make groupby object with pandas
+        groupby_data = df.groupby([group_header])
+
+        for group in "ABCDE":
+            data_from_group = groupby_data.get_group(group)[data_header]
+            # take a stat of the grouped data
+            stat = np.median(data_from_group)
+            # add to dictionary
+            group_stats[group] = stat
+
+        # create violin fig
+        fig = FF.create_violin(df, data_header='Score', group_header='Group',
+                               height=600, width=1000, use_colorscale=True,
+                              group_stats=group_stats)
+
+        # plot
+        py.iplot(fig, filename='Violin Plot with Colorscale')
+        ```
+        """
+        if _numpy_imported is False:
+            raise ImportError("FigureFactory.create_violin() requires "
+                              "numpy to be imported.")
+
+        if _scipy_imported is False:
+            raise ImportError("FigureFactory.create_violin() requires "
+                              "scipy to be imported.")
+
+        from plotly.graph_objs import graph_objs
+        from numbers import Number
```

Comment:
This is standard lib, you should import it at the top of this file. We only want to protect imports when necessary.